h/static/scripts/test/build-thread-test.js

```diff
@@ -262,6 +263,44 @@ describe('build-thread', function () {
         }]);
       });
     });
+
+    context('thread filtering', function () {
+      var fixture = [{
+        id: '1',
+        text: 'annotation',
+        target: [{selector: {}}],
+      },{
+        id: '2',
+        text: 'note',
+        target: [{selector: undefined}],
+      }];
+
+      it('shows only annotations which are notes', function () {
+        var thread = createThread(fixture, {
+          threadFilterFn: function (thread) {
```

Comment:
From the point of view of `buildThread()`, the threadFilterFn is generic - so actually here you only need a single test which checks that only annotations matching the filter are included. eg. `it('shows only annotations matching the thread filter')`. The filter condition could be anything.